### ISSUE TITLE
Basic Image Updatezeit V2.9.50 [bug]: 

### ISSUE THREAD

**Issue by Xsev1** [ioBroker/ioBroker.vis-2#423]
### No existing issues.

- [X] There is no existing issue for my problem.

### Describe the bug

Nach Aktualisierung von 2.9.32 auf 2.9.50 funktioniert die Updatezeit (refresh time) beim Basic Image nicht mehr. Bin jetzt wieder auf 2.9.32 zurück, dort funktioniert das aktualisieren tadellos.

### To Reproduce

-

### Expected behavior

-

### Screenshots & Logfiles

-

### Adapter version

2.9.50

### js-controller version

5.0.19

### Node version

18.20.2

### Operating system

Debian 12

**Comment by tmikulski** [comment]
I have the same issue (including the current version 2.10.7)

**Comment by iAgrippa** [comment]
ja - das kann ich auch bestätigen. Hatte das gleiche Verhalten bei Basic Image aber auch bei Basic Image 8.

**Comment by peterfido** [comment]
Auch unter 2.9.64 keine Besserung. Zurück auf 2.9.32 geht zwar, allerdings fehlen da doch viele Widgets wie Basic HTML navigation

**Comment by jolichter** [comment]
Hello, I can confirm this issue and have experienced it with version 2.9.64 as well.

In VIS2, I integrated a Basic Image widget for a camera using the snapshot URL of my camera. The snapshot URL worked perfectly with version 2.9.32, and I received an updated image every 2 seconds since I had set the update interval (ms) to 2000 for reasonable image refresh rates.

![basic_refreshinterval](https://github.com/user-attachments/assets/334c4a07-c18f-4846-8d46-ddab74fcda93)


Tested with Fully Browser 1.53, Google Chrome 131.0.6778.86, Brave 1.73.91, and ioBroker Visu 1.3.0.

**Comment by TotKur** [comment]
Hello, I can confirm this issue and have experienced it with version 2.11.1 as well.

**Comment by jolichter** [comment]
Yes, even with version 2.11.1, the bug persists. My small [script](https://github.com/jolichter/visWidgetUpdater) could serve as a workaround until it gets fixed.

**Comment by TotKur** [comment]
Hallo,

 

danke für das Script – funktioniert bei mir, aber leider nicht vollständig …

1.	Script in Vis2 unter Scripte hinzugeführt
2.	Widgets ID angepasst – auf das letzte (nicht) Komma berücksichtigt
3.	Updatezeit im Widget auf 0
4.	Logging auf true

Das Resultat:

1.	Keine Errors im Log File
2.	Verwende ich den Link zum go2rtc frame erfolgt die Aktualisierung wie gewünscht.
3.	Verwende ich den link direkt zur Reolink WebCam wird das Bild (im Image aus Vis2 Basic) unmittelbar sichtbar, verschwindet aber nach der ersten Aktualisierung ???

a.	Ist hier eventuell die fehlende „Addiere zu URL“ verantwortlich ??

4.	Beide links funktioniert parallel aus den Browsern Chrome und Edge einwandfrei

 

Ersuche um Info.

 

Danke und lg, Kurt

 

Von: Jo Lichter ***@***.***> 
Gesendet: Freitag, 24. Januar 2025 21:53
An: ioBroker/ioBroker.vis-2 ***@***.***>
Cc: TotKur ***@***.***>; Comment ***@***.***>
Betreff: Re: [ioBroker/ioBroker.vis-2] Basic Image Updatezeit V2.9.50 [bug]: (Issue #423)

 

Yes, even with version 2.11.1, the bug persists. My small script <https://github.com/jolichter/visWidgetUpdater>  could serve as a workaround until it gets fixed.

**Comment by jolichter** [comment]
Wenn dein Bild nur einmal sichtbar ist und dann verschwindet, wird vermutlich etwas Falsches geladen. Du kannst das eventuell über die Konsole (F12) in deinem Browser überprüfen.

Bei meiner Instar-Kamera und dem Regenradar-Bild funktioniert es ohne Probleme. Mit go2rtc oder Reolink habe ich keine Erfahrung, daher kann ich das nicht reproduzieren. Möglicherweise liegt es daran, wie das Passwort in der URL integriert wird. Bei mir sieht das zum Beispiel so aus:
`http://192.168.1.51:80/tmpfs/auto.jpg?usr=USERNAME&pwd=PASSWORT`

Was meinst du genau mit „Ist hier eventuell die fehlende ‚Addiere zu URL‘ verantwortlich“? Diese Checkbox ist bei mir deaktiviert.

PS: Teste das Regenradar doch einmal per Basic Image mit dieser URL:
`https://morgenwirdes.de/api/v3/gif5.php?plz=48143`
Setze den Intervall des witgets im script testweise auf 4000. Dadurch kannst du erkennen, dass das GIF alle 4 Sekunden neu geladen wird. Das ist zwar nicht sinnvoll, dient jedoch als guter Test.

**Comment by lenusch** [comment]
Bei mir auch, heute wieder 2 Stunden alles mögliche getestet, ws socket.io kram.
Einzige abhilfe: Downgrade VIS-2 auf 2.9.32 .....

**Comment by lenusch** [comment]
Finde ich auch schade. Wieder paar Stunden alles mögliche getestet. geupdatet.

**Comment by peterfido** [comment]
Vielleicht zu wenige, die das nutzen.

**Comment by peterfido** [comment]
Vielleicht uninterressant, oder der Workaround von Jolichter gilt insgeheim als Lösung. Vielleicht nochmal im iobroker - Forum ins Gedächtnis holen.

**Comment by oweitman** [comment]
Hi Denis, hier meine Analyse im Forum
https://forum.iobroker.net/topic/82829/was-muss-man-tun-um-einen-bugfix-zu-bekommen/3?_=1762711537893

ich habe mir das mal angeschaut.
Leider ist die Lösung etwas komplexer, da das Problem nicht direkt in vis2 liegt sondern an einer verwendeten bibliothek adapter-react-v5.

Das verwendete React-Element Icon (aus adapter-react-v5) wendet das ref-Element nicht korrekt an (es muss forwardRef verwendet werden).
Daher ist dann im Image Element der Verweis auf das DOM-Element immer null.
Daher wird der refresh nie ausgeführt.

Da adapter-react-v5 relativ zentral bei allen React Anwendungen des iobrokers ist, ist dort eine Änderung gefährlich, da sie Auswirkung auf alles haben kann.

Das muss dann tatsächlich bluefox/denis machen.

<img width="1055" height="920" alt="Image" src="https://github.com/user-attachments/assets/d2a9d3c7-c5f4-4722-ad79-3f9652373d63" />

**Comment by heiko73** [comment]
Habe die 2.13.7 installiert und bei mir werden die Images leider nicht aktualisiert. Danke auf jeden Fall, dass Ihr Euch da jetzt drum kümmert!

**Comment by heiko73** [comment]
> Das Tool von jolichter funktioniert bei mir nicht, auch wenn es bei einigen Leuten zu funktionieren scheint. Es ist auch mir schleierhaft, dass man solch einen Bug nicht löst. Da können sie auch gleich die Updatezeile rausnehmen, das wärer ehrlicher, Das ist der Grund, weshalb ich produktiv immer noch bei VIS-1 bin. Und solange meine Kameras nicht laufen, wird das auch so bleiben. Wenn mit "Next Generation Editor" wohl eher "Next Bug Editor" gemeint ist, dann wird es wohl auch so bleiben.

Bei mir funktioniert das Script tatsächlich auch sehr gut und ist ein akzeptabler Workaround.
